src/QuadForm/Spaces.jl

```diff
@@ -654,6 +654,23 @@ function orthogonal_complement(V::AbsSpace, M::MatElem)
   return K
 end
 
+@doc Markdown.doc"""
+    orthogonal_projection(V::AbsSpace, M::T) where T <: MatElem -> AbsSpaceMor
+
+Given a space `V` and a non-degenerate subspace `W` with basis matrix `M`,
+return the endomorphism of `V` corresponding to the projection onto the
+complement of `W` in `V`.
+"""
+function orthogonal_projection(V::AbsSpace, M::MatElem)
+  _Q = inner_product(V, M, M)
+  @req rank(_Q) == nrows(_Q) "Subspace must be non-degenerate for the inner product on V"
+  U = orthogonal_complement(V, M)
+  B = vcat(U, M)
+  p = vcat(U, zero(M))
+  pr = inv(B)*p
```

Comment:
This is already done in the definition of `p`: if I put the identity instead of the copy of `U` and then I multiply `pr` on the right by `B` I will obtain the same result.

Reply:
O.k. I see it now. Thank you.